test/functional/readpreference.test.js

```diff
@@ -655,3 +656,77 @@ describe('ReadPreference', function() {
     });
   });
 });
+
+describe.only('ReadPreference - operations', () => {
+  const COLLECTION = 'ddl_collection';
+  beforeEach(function(done) {
+    const configuration = this.configuration;
+    const client = configuration.newClient(configuration.writeConcernMax(), {
+      useUnifiedTopology: true,
+      readPreference: 'primaryPreferred'
+    });
+    client.connect((err, client) => {
```

Comment:
all of the methods in this function seem to ignore the possibility for an `err`, except for `connect`. Consider using `withClient` to ensure client is cleaned up, and adding assertions for the non-existence of `err`.